lib/mzgl/util/Dialogs.cpp

```diff
@@ -822,7 +826,73 @@ void Dialogs::launchUrlInWebView(std::string url, std::function<void()> completi
 																	 animated:YES
 																   completion:nil];
 #	else // mac
-
+	dispatch_async(dispatch_get_main_queue(), ^{
+	  NSWindow *win	   = (__bridge NSWindow *) app.windowHandle;
```

Comment:
That fact this section is so long makes me think it could be split up in to different functions, with functional naming to make it simpler to read